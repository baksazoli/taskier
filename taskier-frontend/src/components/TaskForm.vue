<template>
  <v-card>
    <v-card-title>{{ isEditing ? "Edit Task" : "New Task" }}</v-card-title>
    <v-card-text>
      <v-form ref="form" @submit.prevent="submit" v-model="isFormValid">
        <v-select
          v-model="localTask.user_id"
          label="User"
          :items="users"
          item-title="name"
          item-value="id"
          required
        ></v-select>
        <v-text-field
          data-test="description-field"
          v-model="localTask.description"
          label="Description"
          required
          :rules="[rules.required, rules.minLength]"
        />
        <v-text-field
          v-model="localTask.estimated_time"
          label="Estimated time"
          type="number"
          required
          :rules="[rules.required, rules.minValue]"
        />
        <v-text-field
          v-model="localTask.used_time"
          label="Used time"
          type="number"
          required
          :rules="[rules.cannotBeNegative]"
        />
        <v-card-actions class="d-flex justify-end">
          <v-btn color="red" @click="emit('close')">Cancel</v-btn>
          <v-btn type="submit" color="primary" :disabled="!isFormValid">{{ isEditing ? "Update" : "Save" }}</v-btn>
        </v-card-actions>
      </v-form>
    </v-card-text>
  </v-card>
</template>

<script setup lang="ts">
import { ref, onMounted, defineProps, defineEmits, computed, inject } from "vue";
import { taskService } from "@/services/taskService";
import { userService } from "@/services/userService";

const snackbar = inject("snackbar");

const props = defineProps({
  task: Object,
});
const emit = defineEmits(["close", "saved"]);

const localTask = ref({
  id: props.task?.id || null,
  description: props.task?.description || "",
  estimated_time: props.task?.estimated_time || 0,
  used_time: props.task?.used_time || 0,
  user_id: props.task?.user_id || null,
});

const users = ref<{ id: number; name: string }[]>([]);
const form = ref();
const isFormValid = ref(false);

onMounted(async () => {
  const response = await userService.getUsers();
  users.value = response.map(user => ({
    id: user.id,
    name: user.name
  }));

  if (!localTask.value.user_id && users.value.length > 0) {
    localTask.value.user_id = users.value[0].id;
  }
});

const isEditing = computed(() => !!props.task?.id);

const rules = {
  required: (value: any) => !!value || "This field is required",
  minLength: (value: string) => (value.length >= 5) || "Description must be at least 5 characters",
  minValue: (value: number) => (value >= 1) || "Estimated time must be at least 1",
  cannotBeNegative: (value: number) => (value >= 0) || "Used time cannot be negative"
};

const submit = async () => {
  const { valid } = await form.value.validate();

  if (!valid) {
    console.warn("Form validation failed");
    snackbar?.show("Please fill out the form correctly.", "error");
    return;
  }

  localTask.value.estimated_time = parseFloat(localTask.value.estimated_time.toString()) || 0;
  localTask.value.used_time = parseFloat(localTask.value.used_time.toString()) || 0;

  try {
    if (isEditing.value) {
      await taskService.update(localTask.value.id, localTask.value);
      snackbar?.show("Task updated successfully!", "success");
    } else {
      await taskService.create(localTask.value);
      snackbar?.show("Task created successfully!", "success");
    }

    emit("saved");
    emit("close");
  } catch (error) {
    console.error("Task saving failed", error);
    snackbar?.show("Failed to save the task.", "error");
  }
};
</script>
