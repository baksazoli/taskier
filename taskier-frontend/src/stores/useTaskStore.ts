import { defineStore } from "pinia";
import { ref, computed } from "vue";
import { taskService } from "@/services/TaskService";
import type { Task } from "@/models/Task";

export const useTaskStore = defineStore("task", () => {
  const tasks = ref<Task[]>([]);

  const fetchTasks = async () => {
    tasks.value = await taskService.getAll();
  };

  const deleteTask = async (taskIds: number[]) => {
    await taskService.delete(taskIds);
    await fetchTasks();
  };

  const updateTask = async (taskId: number, data: Partial<Task>) => {
    await taskService.update(taskId, data);
    await fetchTasks();
  };

  const bulkDeleteTasks = async (taskIds: number[]) => {
    await taskService.bulkDeleteTasks(taskIds);
    await fetchTasks();
  };

  const bulkCompleteTasks = async (taskIds: number[]) => {
    await taskService.bulkCompleteTasks(taskIds);
    await fetchTasks();
  };

  const totalTime = computed(() => {
    return {
      used: tasks.value.reduce((sum, task) => sum + (task.used_time || 0), 0),
      estimated: tasks.value.reduce((sum, task) => sum + (task.estimated_time || 0), 0),
    };
  });

  return {
    tasks,
    fetchTasks,
    deleteTask,
    updateTask,
    bulkDeleteTasks,
    bulkCompleteTasks,
    totalTime,
  };
});