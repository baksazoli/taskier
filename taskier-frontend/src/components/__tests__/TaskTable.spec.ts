import { mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';
import { useTaskStore } from '@/stores/useTaskStore';
import TaskTable from '@/components/TaskTable.vue';
import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
import { createVuetify } from 'vuetify';
import * as components from 'vuetify/components';
import * as directives from 'vuetify/directives';

const vuetify = createVuetify({ components, directives });

if (typeof ResizeObserver === 'undefined') {
  global.ResizeObserver = class ResizeObserver {
    callback: any;
    constructor(callback: any) {
      this.callback = callback;
    }
    observe() {}
    unobserve() {}
    disconnect() {}
  };
}

describe('TaskTable', () => {
  let wrapper;
  let store;

  beforeEach(async () => {
    const pinia = createTestingPinia({
      createSpy: vi.fn,
      stubActions: false,
    });

    store = useTaskStore();
    store.tasks = [
      { id: 1, description: 'Task 1', estimated_time: 10, used_time: 5, completed_at: null },
      { id: 2, description: 'Task 2', estimated_time: 20, used_time: 15, completed_at: '2023-01-01' },
    ];
    store.totalTime = { used: 20, estimated: 30 };
    store.fetchTasks = vi.fn().mockResolvedValue(undefined);
    store.deleteTask = vi.fn().mockResolvedValue(undefined);
    store.updateTask = vi.fn().mockResolvedValue(undefined);

    wrapper = mount(TaskTable, {
      global: {
        plugins: [pinia, vuetify],
        stubs: {
          TaskForm: true,
        },
      },
    });

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    wrapper.unmount()
  });

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders the component correctly', () => {
    expect(wrapper.exists()).toBe(true)
  })

  it('should open modal when New Task button is clicked', async () => {
    const newTaskButton = wrapper.findComponent('[data-test="new-task-button"]');

    expect(newTaskButton.exists()).toBe(true)

    await newTaskButton.trigger('click');

    expect(wrapper.vm.dialog).toBe(true);
  })

  it('filters tasks correctly based on search query', async () => {
    const searchField = wrapper.findComponent({ name: 'v-text-field' });

    await searchField.setValue('Task 1');

    const filteredTasks = wrapper.vm.filteredTasks;
    expect(filteredTasks.length).toBe(1);
    expect(filteredTasks[0].description).toBe('Task 1');
  })

  it('opens edit modal correctly', async () => {
    const openEditModalSpy = vi.spyOn(wrapper.vm, 'openEditModal')
    const editButton = wrapper.findComponent('[data-test="edit-button"]');
    await editButton.trigger('click')
    
    expect(openEditModalSpy).toHaveBeenCalled()
  });

  it('shows the correct snackbar message after task deletion', async () => {
    const snackbarShowSpy = vi.spyOn(wrapper.vm.snackbar, 'show')

    wrapper.vm.snackbar.show('Task deleted successfully', 'success')
    
    expect(snackbarShowSpy).toHaveBeenCalledWith('Task deleted successfully', 'success')
  })

  it('executes complete task action correctly', async () => {
    const bulkCompleteTasksSpy = vi.fn()
    
    store.bulkCompleteTasks = bulkCompleteTasksSpy;
    
    wrapper.vm.actionType = 'bulkCompleteTasks'
    wrapper.vm.selected = [1, 2]
    
    await wrapper.vm.executeAction()
    
    expect(bulkCompleteTasksSpy).toHaveBeenCalledWith([1, 2])
    expect(wrapper.vm.selected).toEqual([])
  })

  it('displays tasks from store', () => {
    expect(wrapper.text()).toContain('Task 1');
    expect(wrapper.text()).toContain('Task 2');
  });

  it.skip('shows correct total time', async () => {
    wrapper.vm.selected.value = [1];
    wrapper.vm.isSelectionEmpty = false;
    await wrapper.vm.$nextTick();
  
    const $usedEstimatedContainer = wrapper.find('[data-test="used-estimated-container"]');
    expect($usedEstimatedContainer.exists()).toBe(true);
    expect($usedEstimatedContainer.text()).toContain('Time used/estimated for selected tasks: 20 / 30 minutes');
  });

  describe('Delete Action', () => {
    it('should show the correct confirm message when bulk delete is triggered', async () => {
      const bulkDeleteButton = wrapper.findComponent({
        name: 'v-btn',
        props: {
          density: 'compact',
          variant: 'plain',
          size: 'large'
        }
      });
      
      wrapper.vm.selected = [1, 2];
      await bulkDeleteButton.trigger('click');
      await wrapper.vm.confirmAction('bulkDeleteTasks');
      const confirmMessage = wrapper.vm.confirmMessage;
      expect(confirmMessage).toBe('Are you sure you want to delete 2 tasks?');
    });
  
    it('executes bulk delete action correctly', async () => {
      const bulkDeleteTasksSpy = vi.fn();
  
      store.bulkDeleteTasks = bulkDeleteTasksSpy;
  
      wrapper.vm.actionType = 'bulkDeleteTasks';
      wrapper.vm.selected = [1, 2];
  
      await wrapper.vm.executeAction();
      expect(bulkDeleteTasksSpy).toHaveBeenCalledWith([1, 2]);
      expect(wrapper.vm.selected).toEqual([]);
    });

    it.skip('calls delete when confirmed', async () => {
      wrapper.vm.selected = [1]; 
      await wrapper.vm.$nextTick();
    
      await wrapper.findComponent('[data-testid="bulk-delete-button"]').trigger('click');
      await wrapper.vm.$nextTick();
    
      const confirmBtn = wrapper.find('[data-testid="confirm-action-btn"]');
      expect(confirmBtn.exists()).toBe(true);
    
      await confirmBtn.trigger('click');
      await wrapper.vm.$nextTick();
    
      expect(store.deleteTask).toHaveBeenCalledWith([1]);
    });
    
  });
});