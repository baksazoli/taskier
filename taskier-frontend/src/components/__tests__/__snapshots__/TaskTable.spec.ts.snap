// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TaskTable > renders correctly 1`] = `
"<div data-v-83bb75ec="" class="v-card v-card--flat v-theme--light bg-transparent v-card--density-default v-card--variant-elevated">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <!-- Header Section -->
  <div data-v-83bb75ec="" class="sticky-header">
    <header data-v-83bb75ec="" class="v-toolbar v-toolbar--flat v-toolbar--density-default bg-transparent elevation-1 v-theme--light v-locale--is-ltr">
      <!---->
      <div class="v-toolbar__content">
        <!---->
        <!---->
        <div data-v-83bb75ec="" class="v-container v-locale--is-ltr px-0 mx-5">
          <div data-v-83bb75ec="" class="v-row align-center justify-center">
            <div data-v-83bb75ec="" class="v-col-md-2 v-col-12"><button data-v-83bb75ec="" type="button" class="v-btn v-theme--light text-primary v-btn--density-default v-btn--size-default v-btn--variant-text" data-test="new-task-button"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span><span class="v-btn__prepend"><i class="mdi-plus mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span><span class="v-btn__content" data-no-activator=""> New Task </span>
                <!---->
                <!---->
              </button></div>
            <div data-v-83bb75ec="" class="v-col-md-6 v-col-12">
              <div data-v-83bb75ec="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-text-field">
                <!---->
                <div class="v-input__control">
                  <div class="v-field v-field--appended v-field--center-affix v-field--variant-filled v-theme--light v-locale--is-ltr">
                    <div class="v-field__overlay"></div>
                    <div class="v-field__loader">
                      <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                        <!---->
                        <div class="v-progress-linear__background"></div>
                        <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                        <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                          <div class="v-progress-linear__indeterminate">
                            <div class="v-progress-linear__indeterminate long"></div>
                            <div class="v-progress-linear__indeterminate short"></div>
                          </div>
                        </transition-stub>
                        <!---->
                      </div>
                    </div>
                    <!---->
                    <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-1">
                        <!---->Search by description
                      </label><label class="v-label v-field-label" for="input-1">
                        <!---->Search by description
                      </label>
                      <!----><input size="1" type="text" id="input-1" aria-describedby="input-1-messages" dense="" class="v-field__input" value="">
                      <!---->
                    </div>
                    <!---->
                    <div class="v-field__append-inner">
                      <!----><i class="mdi-magnify mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i>
                    </div>
                    <div class="v-field__outline">
                      <!---->
                      <!---->
                    </div>
                  </div>
                </div>
                <!---->
                <div id="input-1-messages" class="v-input__details" role="alert" aria-live="polite">
                  <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages">
                    <!---->
                  </transition-group-stub>
                  <!---->
                </div>
              </div>
            </div>
            <div data-v-83bb75ec="" class="v-col-md-4 v-col-12">
              <div data-v-83bb75ec="" class="v-input v-input--horizontal v-input--center-affix v-input--density-default v-theme--light v-locale--is-ltr v-input--dirty v-text-field v-select v-select--single v-select--selected">
                <!---->
                <div class="v-input__control">
                  <div class="v-field v-field--active v-field--appended v-field--center-affix v-field--dirty v-field--variant-filled v-theme--light v-locale--is-ltr" role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-controls="v-menu-5">
                    <div class="v-field__overlay"></div>
                    <div class="v-field__loader">
                      <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                        <!---->
                        <div class="v-progress-linear__background"></div>
                        <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                        <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                          <div class="v-progress-linear__indeterminate">
                            <div class="v-progress-linear__indeterminate long"></div>
                            <div class="v-progress-linear__indeterminate short"></div>
                          </div>
                        </transition-stub>
                        <!---->
                      </div>
                    </div>
                    <!---->
                    <div class="v-field__field" data-no-activator=""><label class="v-label v-field-label v-field-label--floating" aria-hidden="true" for="input-3">
                        <!---->Filter by completion
                      </label><label class="v-label v-field-label" for="input-3">
                        <!---->Filter by completion
                      </label>
                      <!---->
                      <div class="v-field__input" data-no-activator="">
                        <!---->
                        <!---->
                        <div class="v-select__selection"><span class="v-select__selection-text">All<!----></span></div><input size="1" type="text" id="input-3" aria-describedby="input-3-messages" inputmode="none" aria-label="Open" title="Open" dense="" value="all">
                      </div>
                      <!---->
                    </div>
                    <!---->
                    <div class="v-field__append-inner">
                      <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-select__menu-icon" aria-hidden="true"></i>
                      <!---->
                    </div>
                    <div class="v-field__outline">
                      <!---->
                      <!---->
                    </div>
                  </div>
                </div>
                <!---->
                <div id="input-3-messages" class="v-input__details" role="alert" aria-live="polite">
                  <transition-group-stub name="slide-y-transition" tag="div" appear="false" persisted="false" css="true" class="v-messages">
                    <!---->
                  </transition-group-stub>
                  <!---->
                </div>
              </div>
            </div>
          </div>
        </div>
        <!---->
      </div>
      <transition-stub name="expand-transition" appear="false" persisted="false" css="true">
        <!---->
      </transition-stub>
    </header>
  </div>
  <div data-v-83bb75ec="" style="height: 1px; padding-top: 10px;">
    <!--v-if-->
  </div><!-- Table Section -->
  <div data-v-83bb75ec="" class="table-container mt-12">
    <div data-v-83bb75ec="" class="v-table v-table--fixed-height v-table--fixed-header v-table--has-top v-table--has-bottom v-theme--light v-table--density-default v-data-table v-data-table--show-select">
      <!---->
      <div class="v-table__wrapper" style="height: calc(100vh - 250px);">
        <table>
          <!---->
          <thead>
            <tr>
              <th class="v-data-table__td v-data-table-column--no-padding v-data-table-column--align-start v-data-table__th v-data-table__th--sticky" style="width: 48px; position: sticky; top: calc(var(--v-table-header-height) * 0);" colspan="1" rowspan="1">
                <div class="v-selection-control v-selection-control--density-default v-checkbox-btn">
                  <div class="v-selection-control__wrapper">
                    <!---->
                    <div class="v-selection-control__input"><i class="mdi-checkbox-blank-outline mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i><input id="input-6" aria-disabled="false" type="checkbox" value="true"></div>
                  </div>
                  <!---->
                </div>
              </th>
              <th class="v-data-table__td v-data-table-column--align-start v-data-table__th--sortable v-data-table__th v-data-table__th--sticky" style="position: sticky; top: calc(var(--v-table-header-height) * 0);" colspan="1" rowspan="1">
                <div class="v-data-table-header__content"><span>Created At</span><i class="mdi-arrow-up mdi v-icon notranslate v-theme--light v-icon--size-default v-data-table-header__sort-icon" aria-hidden="true"></i>
                  <!---->
                </div>
              </th>
              <th class="v-data-table__td v-data-table-column--align-start v-data-table__th--sortable v-data-table__th v-data-table__th--sticky" style="position: sticky; top: calc(var(--v-table-header-height) * 0);" colspan="1" rowspan="1">
                <div class="v-data-table-header__content"><span>Assignee</span><i class="mdi-arrow-up mdi v-icon notranslate v-theme--light v-icon--size-default v-data-table-header__sort-icon" aria-hidden="true"></i>
                  <!---->
                </div>
              </th>
              <th class="v-data-table__td v-data-table-column--align-start v-data-table__th--sortable v-data-table__th v-data-table__th--sticky" style="position: sticky; top: calc(var(--v-table-header-height) * 0);" colspan="1" rowspan="1">
                <div class="v-data-table-header__content"><span>Description</span><i class="mdi-arrow-up mdi v-icon notranslate v-theme--light v-icon--size-default v-data-table-header__sort-icon" aria-hidden="true"></i>
                  <!---->
                </div>
              </th>
              <th class="v-data-table__td v-data-table-column--align-start v-data-table__th--sortable v-data-table__th v-data-table__th--sticky" style="position: sticky; top: calc(var(--v-table-header-height) * 0);" colspan="1" rowspan="1">
                <div class="v-data-table-header__content"><span>Estimated Time (minute)</span><i class="mdi-arrow-up mdi v-icon notranslate v-theme--light v-icon--size-default v-data-table-header__sort-icon" aria-hidden="true"></i>
                  <!---->
                </div>
              </th>
              <th class="v-data-table__td v-data-table-column--align-start v-data-table__th--sortable v-data-table__th v-data-table__th--sticky" style="position: sticky; top: calc(var(--v-table-header-height) * 0);" colspan="1" rowspan="1">
                <div class="v-data-table-header__content"><span>Used Time (minute)</span><i class="mdi-arrow-up mdi v-icon notranslate v-theme--light v-icon--size-default v-data-table-header__sort-icon" aria-hidden="true"></i>
                  <!---->
                </div>
              </th>
              <th class="v-data-table__td v-data-table-column--align-start v-data-table__th--sortable v-data-table__th v-data-table__th--sticky" style="position: sticky; top: calc(var(--v-table-header-height) * 0);" colspan="1" rowspan="1">
                <div class="v-data-table-header__content"><span>Completed</span><i class="mdi-arrow-up mdi v-icon notranslate v-theme--light v-icon--size-default v-data-table-header__sort-icon" aria-hidden="true"></i>
                  <!---->
                </div>
              </th>
              <th class="v-data-table__td v-data-table-column--align-start v-data-table__th v-data-table__th--sticky" style="position: sticky; top: calc(var(--v-table-header-height) * 0);" colspan="1" rowspan="1">
                <div class="v-data-table-header__content"><span>Actions</span>
                  <!---->
                  <!---->
                </div>
              </th>
            </tr>
            <!---->
          </thead>
          <!---->
          <tbody>
            <!---->
            <tr class="v-data-table__tr">
              <td class="v-data-table__td v-data-table-column--no-padding v-data-table-column--align-start v-data-table__td--select-row" style="width: 48px;">
                <div class="v-selection-control v-selection-control--density-default v-checkbox-btn">
                  <div class="v-selection-control__wrapper">
                    <!---->
                    <div class="v-selection-control__input"><i class="mdi-checkbox-blank-outline mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i><input id="input-7" aria-disabled="false" type="checkbox" value="true"></div>
                  </div>
                  <!---->
                </div>
              </td>
              <td class="v-data-table__td v-data-table-column--align-start">03/30/2025 17:57</td>
              <td class="v-data-table__td v-data-table-column--align-start"></td>
              <td class="v-data-table__td v-data-table-column--align-start">Task 1</td>
              <td class="v-data-table__td v-data-table-column--align-start">10</td>
              <td class="v-data-table__td v-data-table-column--align-start">5</td>
              <td class="v-data-table__td v-data-table-column--align-start"><i data-v-83bb75ec="" class="mdi-close mdi v-icon notranslate v-theme--light v-icon--size-default text-red" aria-hidden="true"></i></td>
              <td class="v-data-table__td v-data-table-column--align-start"><button data-v-83bb75ec="" type="button" class="v-btn v-theme--light v-btn--density-compact v-btn--size-large v-btn--variant-plain" data-test="edit-button" aria-describedby="v-tooltip-9"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
                  <!----><span class="v-btn__content" data-no-activator=""><!----><!--teleport start--><!--teleport end--><i data-v-83bb75ec="" class="mdi-pencil mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
                  <!---->
                  <!---->
                </button><button data-v-83bb75ec="" type="button" class="v-btn v-theme--light v-btn--density-compact v-btn--size-large v-btn--variant-plain" aria-describedby="v-tooltip-11"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
                  <!----><span class="v-btn__content" data-no-activator=""><!----><!--teleport start--><!--teleport end--><i data-v-83bb75ec="" class="mdi-delete mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
                  <!---->
                  <!---->
                </button></td>
            </tr>
            <!---->
            <tr class="v-data-table__tr">
              <td class="v-data-table__td v-data-table-column--no-padding v-data-table-column--align-start v-data-table__td--select-row" style="width: 48px;">
                <div class="v-selection-control v-selection-control--density-default v-checkbox-btn">
                  <div class="v-selection-control__wrapper">
                    <!---->
                    <div class="v-selection-control__input"><i class="mdi-checkbox-blank-outline mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i><input id="input-12" aria-disabled="false" type="checkbox" value="true"></div>
                  </div>
                  <!---->
                </div>
              </td>
              <td class="v-data-table__td v-data-table-column--align-start">03/30/2025 17:57</td>
              <td class="v-data-table__td v-data-table-column--align-start"></td>
              <td class="v-data-table__td v-data-table-column--align-start">Task 2</td>
              <td class="v-data-table__td v-data-table-column--align-start">20</td>
              <td class="v-data-table__td v-data-table-column--align-start">15</td>
              <td class="v-data-table__td v-data-table-column--align-start"><i data-v-83bb75ec="" class="mdi-check mdi v-icon notranslate v-theme--light v-icon--size-default text-green" aria-hidden="true"></i></td>
              <td class="v-data-table__td v-data-table-column--align-start"><button data-v-83bb75ec="" type="button" class="v-btn v-theme--light v-btn--density-compact v-btn--size-large v-btn--variant-plain" data-test="edit-button" aria-describedby="v-tooltip-14"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
                  <!----><span class="v-btn__content" data-no-activator=""><!----><!--teleport start--><!--teleport end--><i data-v-83bb75ec="" class="mdi-pencil mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
                  <!---->
                  <!---->
                </button><button data-v-83bb75ec="" type="button" class="v-btn v-theme--light v-btn--density-compact v-btn--size-large v-btn--variant-plain" aria-describedby="v-tooltip-16"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
                  <!----><span class="v-btn__content" data-no-activator=""><!----><!--teleport start--><!--teleport end--><i data-v-83bb75ec="" class="mdi-delete mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
                  <!---->
                  <!---->
                </button></td>
            </tr>
            <!---->
            <!---->
          </tbody>
          <!---->
          <!---->
        </table>
      </div>
      <hr class="v-divider v-theme--light" aria-orientation="horizontal" role="separator">
      <div class="v-data-table-footer">
        <!---->
        <div class="v-data-table-footer__items-per-page"><span>Items per page:</span>
          <div class="v-input v-input--horizontal v-input--center-affix v-input--density-compact v-theme--light v-locale--is-ltr v-input--dirty v-text-field v-select v-select--single v-select--selected">
            <!---->
            <div class="v-input__control">
              <div class="v-field v-field--active v-field--appended v-field--center-affix v-field--dirty v-field--no-label v-field--variant-outlined v-theme--light v-locale--is-ltr" role="combobox" aria-haspopup="listbox" aria-expanded="false" aria-controls="v-menu-19">
                <div class="v-field__overlay"></div>
                <div class="v-field__loader">
                  <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
                    <!---->
                    <div class="v-progress-linear__background"></div>
                    <div class="v-progress-linear__buffer" style="width: 0%;"></div>
                    <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
                      <div class="v-progress-linear__indeterminate">
                        <div class="v-progress-linear__indeterminate long"></div>
                        <div class="v-progress-linear__indeterminate short"></div>
                      </div>
                    </transition-stub>
                    <!---->
                  </div>
                </div>
                <!---->
                <div class="v-field__field" data-no-activator="">
                  <!---->
                  <!---->
                  <!---->
                  <div class="v-field__input" data-no-activator="">
                    <!---->
                    <!---->
                    <div class="v-select__selection"><span class="v-select__selection-text">50<!----></span></div><input size="1" type="text" id="input-17" aria-describedby="input-17-messages" inputmode="none" aria-label="Open" title="Open" value="50">
                  </div>
                  <!---->
                </div>
                <!---->
                <div class="v-field__append-inner">
                  <!----><i class="mdi-menu-down mdi v-icon notranslate v-theme--light v-icon--size-default v-select__menu-icon" aria-hidden="true"></i>
                  <!---->
                </div>
                <div class="v-field__outline">
                  <div class="v-field__outline__start"></div>
                  <!---->
                  <div class="v-field__outline__end"></div>
                  <!---->
                </div>
              </div>
            </div>
            <!---->
            <!---->
          </div>
        </div>
        <div class="v-data-table-footer__info">
          <div>1-2 of 2</div>
        </div>
        <div class="v-data-table-footer__pagination">
          <nav class="v-pagination v-theme--light" role="navigation" aria-label="Pagination Navigation" data-test="v-pagination-root">
            <ul class="v-pagination__list">
              <li class="v-pagination__first" data-test="v-pagination-first"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-comfortable v-btn--rounded v-btn--size-default v-btn--variant-plain" disabled="" aria-label="First page" aria-disabled="true"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
                  <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-page-first mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
                  <!---->
                  <!---->
                </button></li>
              <li class="v-pagination__prev" data-test="v-pagination-prev"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-comfortable v-btn--rounded v-btn--size-default v-btn--variant-plain" disabled="" aria-label="Previous page" aria-disabled="true"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
                  <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-left mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
                  <!---->
                  <!---->
                </button></li>
              <li class="v-pagination__next" data-test="v-pagination-next"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-comfortable v-btn--rounded v-btn--size-default v-btn--variant-plain" disabled="" aria-label="Next page" aria-disabled="true"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
                  <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-chevron-right mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
                  <!---->
                  <!---->
                </button></li>
              <li class="v-pagination__last" data-test="v-pagination-last"><button type="button" class="v-btn v-btn--disabled v-btn--icon v-theme--light v-btn--density-comfortable v-btn--rounded v-btn--size-default v-btn--variant-plain" disabled="" aria-label="Last page" aria-disabled="true"><span class="v-btn__overlay"></span><span class="v-btn__underlay"></span>
                  <!----><span class="v-btn__content" data-no-activator=""><i class="mdi-page-last mdi v-icon notranslate v-theme--light v-icon--size-default" aria-hidden="true"></i></span>
                  <!---->
                  <!---->
                </button></li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
  </div><!-- Dialogs -->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;
